specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/networkWatcher.json

```diff
@@ -1525,1535 +1525,2590 @@
-        }
-      },
-      "allOf": [
-        {
-          "$ref": "#/definitions/PacketCaptureParameters"
-        }
+    "ConnectivityParameters": {
+      "description": "Parameters that determine how the connectivity check will be performed.",
+      "required": [
+        "source",
+        "destination"
       ],
-      "description": "The properties of a packet capture session."
-    },
-    "PacketCaptureQueryStatusResult": {
       "properties": {
-        "name": {
-          "type": "string",
-          "description": "The name of the packet capture resource."
-        },
-        "id": {
-          "type": "string",
-          "description": "The ID of the packet capture resource."
+        "source": {
+          "$ref": "#/definitions/ConnectivitySource",
+          "description": "The source of the connection."
         },
-        "captureStartTime": {
-          "type": "string",
-          "format": "date-time",
-          "description": "The start time of the packet capture session."
+        "destination": {
+          "$ref": "#/definitions/ConnectivityDestination",
+          "description": "The destination of connection."
         },
-        "packetCaptureStatus": {
+        "protocol": {
           "type": "string",
+          "description": "Network protocol.",
           "enum": [
-            "NotStarted",
-            "Running",
-            "Stopped",
-            "Error",
-            "Unknown"
+            "Tcp",
+            "Http",
+            "Https",
+            "Icmp"
           ],
           "x-ms-enum": {
-            "name": "PcStatus",
+            "name": "Protocol",
             "modelAsString": true
-          },
-          "description": "The status of the packet capture session."
+          }
         },
-        "stopReason": {
-          "type": "string",
-          "description": "The reason the current packet capture session was stopped."
+        "protocolConfiguration": {
+          "$ref": "#/definitions/ProtocolConfiguration",
+          "description": "Configuration of the protocol."
         },
-        "packetCaptureError": {
-          "type": "array",
-          "description": "List of errors of packet capture session.",
-          "items": {
-            "type": "string",
-            "enum": [
-              "InternalError",
-              "AgentStopped",
-              "CaptureFailed",
-              "LocalFileFailed",
-              "StorageFailed"
-            ],
-            "x-ms-enum": {
-              "name": "PcError",
-              "modelAsString": true
-            }
-          }
+        "preferredIPVersion": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "description": "Preferred IP version of the connection."
         }
-      },
-      "description": "Status of packet capture session."
+      }
     },
-    "TroubleshootingParameters": {
-      "description": "Parameters that define the resource to troubl
```

Comment:
What properties in here are required to be provided by the user when specifying a ConnectionMonitorEndpoint (in general make a pass through and mark properties as required when they are required on all connectionMonitor definitions)

Reply:
I went through all properties we have and when it was possible, I specified what is requred.